cumulusci/cli/cci.py

```diff
@@ -809,13 +809,63 @@ def org_connect(config, org_name, sandbox, login_url, default, global_org):
     oauth_dict = oauth_capture()
     org_config = OrgConfig(oauth_dict, org_name)
     org_config.load_userinfo()
-
-    config.keychain.set_org(org_config, global_org)
+    org_config._load_orginfo()
+    datetime_format = "%Y-%m-%dT%H:%M:%S.%f+0000"
+    if org_config.organization_sobject["IsSandbox"]:
+        # confirmed org is scratch org
+        if org_config.organization_sobject["TrialExpirationDate"]:
+            days = (
+                datetime.strptime(
+                    org_config.organization_sobject["TrialExpirationDate"],
+                    datetime_format,
+                )
+                - datetime.strptime(
+                    org_config.organization_sobject["CreatedDate"], datetime_format
+                )
+            ).days
+            # need to handle this case!
+            if org_config.organization_sobject["NamespacePrefix"]:
+                org_config.config.update(
+                    {"namespaced": org_config.organization_sobject["NamespacePrefix"]}
+                )
```

Comment:
`namespaced` is a boolean based on whether the org has a namespace; you can just do `"namespaced": bool(org_config.organization_sobject["NamespacePrefix"])`